api/src/main/java/org/apache/iceberg/ContentFile.java

```diff
@@ -192,4 +192,12 @@ default F copyWithStats(Set<Integer> requestedColumnIds) {
   default F copy(boolean withStats) {
     return withStats ? copy() : copyWithoutStats();
   }
+
+  /**
+   * Returns the path of the manifest which this file is referenced in or null if it was not read
+   * from a manifest.
+   */
+  default CharSequence manifestPath() {
```

Comment:
Shall we put this method before `pos()` so both are co-located?

Reply:
We need to discuss using `CharSequence` for paths in our APIs. I am not sure this approach is working out very well. I believe the idea was to expose Avro `Utf8` directly. The problem is that `Utf8` is not serializable. Therefore, we always convert `Utf8` to `String`. Working with `CharSequence` adds a lot of complexity for no performance benefit. Puffin and a few other places use `String` for paths.

@rdblue, is my assumption about `CharSequence` correct? What are your thoughts now?

Reply:
BTW, `path()` in `ManifestFile` returns `String`.

Reply:
I think it would be fine to use String here and always expect to convert.

Reply:
Updated to use String. I also published https://github.com/apache/iceberg/pull/11092 to deprecate the path API which returns CharSequence, in favor of a `location` API which returns String. This removal would happen in 2.0